examples/operator.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: etcd-cluster-operator
+  namespace: default
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  namespace: default
+  name: configmap-reader
+rules:
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get", "create", "watch", "update"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["create", "get", "list", "watch"]
+  - apiGroups: ["apps"]
+    resources: ["replicasets"]
+    verbs: ["list", "get", "create", "watch"]
+  - apiGroups: ["etcd.improbable.io"]
+    resources: ["etcdpeers"]
+    verbs: ["list", "get", "watch"]
+  - apiGroups: ["etcd.improbable.io"]
+    resources: ["etcdclusters"]
+    verbs: ["list", "get", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: etcd-cluster-operator-configmap-reader
+  namespace: default
+subjects:
+  - kind: ServiceAccount
+    name: etcd-cluster-operator
+    namespace: default
+roleRef:
+  kind: ClusterRole
+  name: configmap-reader
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: etcd-cluster-operator
+  name: etcd-cluster-operator
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: etcd-cluster-operator
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: etcd-cluster-operator
+    spec:
+      containers:
+      - args:
+        - --enable-leader-election
+        command:
+        - /manager
+        image: etcd-cluster-operator:test
+        imagePullPolicy: IfNotPresent
+        name: operator
+        resources:
+          limits:
+            cpu: 100m
+            memory: 50Mi
+          requests:
+            cpu: 100m
+            memory: 20Mi
+        terminationMessagePath: /dev/termination-log
+        terminationMessagePolicy: File
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      terminationGracePeriodSeconds: 10
+      serviceAccountName: etcd-cluster-operator
```

Comment:
How will this file be kept up to date? Can the config in `config/` be used instead, with an additional overlay to set `image` and `imagePullPolicy` appropriately?

Reply:
imo we shouldn't be maintaining both `examples/` and `config/`, as there's a lot of duplication between them. i'd prefer if we chose one and consistently used it (for both docs and test fixtures). for me i'd lean towards `examples/` as i find them a bit more easy to use, and i'm not sure we're getting many benefits out of kustomize for this relatively small set of resources, but we would lose some of the kubebuilder codegen for CRDs.

thoughts? cc @JamesLaverack @wallrj

Reply:
The `examples` file was really just a scratch-pad I made to begin with. If we want to keep only one I'd prefer `config` purely because it's auto-generated.

I'm fine with Kustomize, I think it's one of those 'extra bits' that Kubebuilder brings that we're not going to know that we need until it's too late. Plus it's quite lightweight now that it's actually built into `kubectl`, so no extra tool required or anything.

Reply:
Looking further into `config` I definitely think we should update and use that. There's already a `config/samples` directory that we can update. We need to clean out `config/certmanager` from the boilerplate code too.

However, this *could* be done in a follow-up PR, with the code using `examples/` for now and the follow-up moving over to `config/`.

Reply:
sgtm. the issue i ran in to with some of the kustomize samples is that its not trivial to provide an override to the pullpolicy/image tag without invoking more shell calls to `kustomize`, and the defaults wont work with the e2e tests. i guess i could also hack around this by applying some patch after creating the deployment from kustomize, or deploying some mutating webhook before.

either way i think it's best to solve this in this PR to avoid introducing more bloat to `examples/` if we don't want to maintain it

Reply:
> i think it's best to solve this in this PR to avoid introducing more bloat to examples/ if we don't want to maintain it

I'm fine with that if you want to take that approach.